lib/moped/protocol/reply.rb

```diff
@@ -108,7 +108,12 @@ def query_failed?
       # @since 1.2.10
       def unauthorized?
         result = documents[0]
-        result["code"] == UNAUTHORIZED || result["assertionCode"] == UNAUTHORIZED
+        return false if result.nil?
+        
+        err = error_message(result)
+        UNAUTHORIZED.include?(result["code"]) ||
```

Comment:
Maybe just add that url in a comment or call out the original upstream file + function it was copy pasta'd from for reference, at most.  I don't think it's undecipherable as-is.